FUNCTIONS.md

```diff
@@ -0,0 +1,161 @@
+|:---------------------------:|:------------:|
+| app: ui, env: staging       | 8   NaN 2    |
+| app: ui, env: production    | 8   6   NaN  |
+| app: server, env: staging   | NaN 9   NaN  |
+| app: server, env: production| 8   3   8    |
+
+The result of `aggregate.mean` is:
+
+|`aggregate.mean(MetricB)` TagSet |`aggregate.mean(MetricB)` Values|
+|:-------------------------------:|:------------------------------:|
+| (no tags)                       | 9 6 5                          |
+
+Aggregations can be grouped by individual tags. The series in the resulting series list preserve those tags which their group used.
+
+|MetricA TagSet               |MetricA Values|
+|:---------------------------:|:------------:|
+| app: ui, env: staging       | 1 2 1        |
+| app: ui, env: production    | 3 3 3        |
+| app: server, env: staging   | 0 0 1        |
+| app: server, env: production| 2 2 0        |
+
+|`aggregate.sum(MetricA group by app)` TagSet|`aggregate.sum(MetricA group by app)` Values|
+|:------------------------------------------:|:------------------------------------------:|
+| app: ui                                    | 4 5 4                                      |
+| app: server                                | 2 2 1                                      |
+
+## Transformation Functions
+
+Transformation functions modify each series in a given serieslist independently. The following transformation functions are supported:
+
+* `transform.derivative`
+* `transform.integral`
+* `transform.rate`
+* `transform.cumulative`
+* `transform.default`
+* `transform.abs`
+* `transform.nan_keep_last`
+
+### `transform.derivative(list)`
+
+This function computes a numerical derivative for a given timeseries. Each value will be updates to be `y[i] = (x[i] - x[i-1]) / interval` where `interval` is the time between samples, measured in seconds.
+The first value is assigned 0.
+
+### `transform.integral(list)`
+
+This function computes a numerical integral for a given timeseries. Each value will be the sum of the values up to it (including itself), a left Riemann integral.
+This quantity will be scaled, interpreting each value in the series as having units of `events / second`, and producing something with the units of `events`.
+If you do not want this behavior, use `transform.cumulative` instead, which does not have scaling behavior.
+
+`NaN` values are treated as 0.
+
+### `transform.rate(list)`
+
+This function computes the numerical derivative of a given timeseries, as in `transform.derivative`, but it bounds the result to be at least 0.
+This is useful for timeseries which increase until reaching some cap where they reset, to avoid massive negative spikes in the resulting derivative.
+It scales the results to be `events / second`, interpreting the input with units `events`.
+
+### `transform.cumulative(list)`
+
+This function computes the cumulative sum of a given timeseries. It performs no scaling. `NaN` values are treated as 0.
```

Comment:
`cumulative sum` doesn't really give too much information since sums are cumulative. We should say something along the lines of this sums the raw values within a timeseries.